src/Pages/Homepage/HomePage.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import { useEffect } from 'react'
+import { Navbar } from '../../Components'
+import axios from "axios";
+import { useCategory } from '../../Context/CategoryContext';
+import { Link } from 'react-router-dom';
+import "./homepage.css"
+
+function HomePage() {
+    const {state,dispatch}=useCategory();
+    
+    useEffect(()=>{
+        (async () => {
+            try {
+                const response = await axios.get("/api/categories");
+                console.log(response.data.categories)
+              dispatch({ type: "SAVE_DATA", payload: response.data.categories });
+            } catch (error) {
+              console.log(error, "Could not load data");
```

Comment:
add some kind of toast library, instead of logs for better interactivity.

Reply:
Thank you done all the changes!